### PR TITLE
Initial impl

### PR DESCRIPTION
**Pull request by gusnaughton**
# Alaska Airlines Pull Request

Adds auro-flight-line. Will update description later.

## Type of change:

Please delete options that are not relevant.

- [x] New capability
- [ ] Revision of an existing capability
- [ ] Infrastructure change (automation, etc.)
- [ ] Other (please elaborate)

## Checklist:

- [x] My update follows the CONTRIBUTING guidelines of this project
- [x] I have performed a self-review of my own update

**By submitting this Pull Request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

_Pull Requests will be evaluated by their quality of update and whether it is consistent with the goals and values of this project. Any submission is to be considered a conversation between the submitter and the maintainers of this project and may require changes to your submission._

**Thank you for your submission!**<br>
-- Auro Design System Team

### CODE REVIEW

**Comment by gusnaughton** [comment]
awaiting design approval from @erikalanghauser

**Comment by gusnaughton** [comment]
@erikalanghauser's requirement for merge is to even out the dots on the flightline.

Current:
image

Figma:
image

I tried setting a margin on .slot-container and extend the ::before's width to 200% to try to get the lines to even out, but this does not equally apply depending on how many elements are in the slot. I'd love to get some advice on this one @blackfalcon 

image

**Review by erikalanghauser — APPROVED** [review]
Design approval granted! ✅

File: src/style-flight-segment.scss [hunk 1]
```diff
@@ -0,0 +1,29 @@
+div {
+    display: inline-flex;
+    align-items: center;
+    justify-content: center;
+    flex-direction: column; 
+    z-index: 2;
+    width: 100%;  
+    
+    &.stopover-container {
+        padding-bottom: 22px;
```
Comment (blackfalcon) [inline src/style-flight-segment.scss:10]: Can this be 

var(--auro-size-lg) | 1.5rem? 24px? That's an Auro spec.

File: src/style-flight-segment.scss [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    border-radius: var(--auro-size-md);
+    width: var(--auro-size-md);
+    height: var(--auro-size-md);
+    &.stopover {
+        background-color: var(--auro-color-background-lightest);
+    }
+}
+
+span {
+    margin-top:14px;
```
Comment (blackfalcon) [inline src/style-flight-segment.scss:26]:
```suggestion
    margin-top: var(--auro-size-sm);
```

File: src/style-flight-segment.scss [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    color: var(--auro-color-text-secondary-on-light);
+    padding-bottom: 6px;
```
Comment (blackfalcon) [inline src/style-flight-segment.scss:28]:
```suggestion
    padding-bottom: var(--auro-size-xs);
```

File: src/style-flightline.scss [hunk 4]
```diff
@@ -27,9 +27,48 @@
...
+.nonstop::before,
+auro-flight-segment::before,
+::slotted(*)::before {
+    content: "";
+    display: block;
+    background-color: #054687;
+    height: var(--auro-size-xxxs);
+    width: 100%;
+    position: relative;
+    top:10px;
```
Comment (blackfalcon) [inline src/style-flightline.scss:47]:
```suggestion
    top: 10px;
```

File: src/style-flightline.scss [hunk 5]
```diff
@@ -27,9 +27,48 @@
 //   ...
 // }
 
-.testClass {
-  color:  var(--auro-color-border-error-on-light);
-  border: 1px solid var(--auro-color-border-error-on-light);
-  display: inline-block;
-  padding: var(--auro-text-body-size-default);
+// forces each segment to take up as much space in its row to extend the ::before
```
Comment (blackfalcon) [inline src/style-flightline.scss:30]: 2 space tabbing, please

File: src/style-flightline.scss [hunk 6]
```diff
@@ -27,9 +27,48 @@
...
+    top:10px;
+    z-index: 1;
+}
+
+// when there's only one connection, we want to show the actual dot but
+// for multiple we want to just show N stops in a layover dot.
+.multiple ::slotted(*) {
+    display: none;
+    @include auro_breakpoint--sm {
+        display:block;
```
Comment (blackfalcon) [inline src/style-flightline.scss:56]:
```suggestion
        display: block;
```

File: src/style-flightline.scss [hunk 7]
```diff
@@ -27,9 +27,48 @@
...
+    }
+}
+
+auro-flight-segment {
+    @include auro_breakpoint--sm {
+        display:none;
```
Comment (blackfalcon) [inline src/style-flightline.scss:62]:
```suggestion
        display: none;
```

File: src/style-flightline.scss [hunk 8]
```diff
@@ -27,9 +27,48 @@
...
+    }
+}
+
+// flex container for the slot
+.slot-container {
+    display:flex;
```
Comment (blackfalcon) [inline src/style-flightline.scss:68]:
```suggestion
    display: flex;
```

File: src/style-flightline.scss [hunk 9]
```diff
@@ -27,9 +27,48 @@
...
+    justify-content:space-around;
```
Comment (blackfalcon) [inline src/style-flightline.scss:69]:
```suggestion
    justify-content: space-around;
```

**Review by blackfalcon — CHANGES_REQUESTED** [review]
Great work, just need to clean up some things.

**Comment by blackfalcon** [comment]
The Rollup config needs to be updated to address the multiple components within the repo. See https://github.com/AlaskaAirlines/auro-icon/blob/master/rollup.config.js for examples. 

Also, be sure to update the README for CDN install instructions.

File: src/auro-flightline.js [hunk 10]
```diff
@@ -5,35 +5,38 @@
...
-  static get properties() {
-    return {
-      // ...super.properties,
-      cssClass:   { type: String }
-    };
+
+  firstUpdated() {
+    // children is an array of auro-flight-segments from within your <slot> below.
+    const children = this.shadowRoot.querySelector('slot').assignedNodes().
+filter((node) => node.nodeName === 'AURO-FLIGHT-SEGMENT');
```
[the remark this answers is not in the document]
Comment (gusnaughton) [reply src/auro-flightline.js:28]: This happened due to the eslint --fix operation. Fixed, but keep an eye out for this in the future.

File: docs/api.md [hunk 11]
```diff
@@ -1,9 +1,32 @@
...
 
 ## Properties
 
-| Property   | Attribute  | Type     | Description                                  |
-|------------|------------|----------|----------------------------------------------|
-| `cssClass` | `cssClass` | `String` | Applies designated CSS class to DOM element. |
+| Property   | Attribute  | Type      | Description                                      |
+|------------|------------|-----------|--------------------------------------------------|
+| `duration` | `duration` | `string`  |                                                  |
+| `iata`     | `iata`     | `String`  | text to display, typically a station code (SEA, PVD, AVP) |
```
[the remark this answers is not in the document]
Comment (gusnaughton) [reply docs/api.md]: This is not accurate - this should be text to display. We use this field to display "2/3/4/5 stops" in responsive mode.

[the remark this answers is not in the document]
Comment (blackfalcon) [reply docs/api.md]: The point I am making is to be clear to the consumer. Yes, we use this to display content, but that's in the logic of the element itself. 

Is there ever a case where a consumer of the element would put in any other string other than an IATA code?

File: src/auro-flightline.js [hunk 12]
```diff
@@ -5,35 +5,37 @@
...
+    const children = this.shadowRoot.querySelector('slot').assignedNodes().
+      filter((node) => node.nodeName === 'AURO-FLIGHT-SEGMENT');
+
+    // if we have a nonstop flight, we need to force _something_ into the dom
+    // since we are using pseudoclasses to draw the connector line.
+    if (!children.length) {
+      const el = document.createElement('span');
+      // hack to get something substantial into the DOM to spawn the ::before
+
+      el.style = 'color: transparent; font-size: 0px;';
```
Comment (blackfalcon) [inline src/auro-flightline.js:35]:
```suggestion
      el.style = 'color: transparent; font-size: 0;';
```

File: demo/demo.md [hunk 13]
```diff
@@ -1,56 +1,170 @@
...
+Auro-Flightline represents a passenger's intermediate journey between a departure station and an arrival station.
 
-Nulla at augue facilisis `odio lobortis` molestie vitae a nulla.
+The flightline represents the layovers and stopovers the guest will take throughout their journey. For instance, flying from SEA (Seattle, WA) to AVP (Scranton, PA) will take a *layover* in ORD (Chicago, IL), while a guest flying from [ANC (Anchorage, AK) to ADK (Adak Island, AK) will have a *stopover* in CDB (Cold Bay, AK)](https://onemileatatime.com/alaska-airlines-adak-cold-bay/). They won't get off the aircraft in Cold Bay, kind of like a bus. We cannot predict stopover duration due to variants in passenger and cargo loads.
 
-## auro-flightline use cases
+Guests with physical disabilities appreciate indication of whether a flight will require them to deplane, and typically optimize for the most direct route. For this reason, [Change of Gauge flights](https://www.travelweekly.com/Mark-Pestronk/Agents-must-identify-change-of-gauge-services), or COG for short, require a *layover* indication even if they are filed as a direct flight. Please contact gus@alaskaair.com for clarification on this.
```
Comment (blackfalcon) [inline demo/demo.md:7]: Delete this. While I appreciate what you are doing here, it is not the purpose of this document to sell a user on the idea. 

Again, if this is something that you feel needs to be expressed, I am all for creating a supporting document that we can link to.

File: src/style-flightline.scss [hunk 14]
```diff
@@ -27,9 +27,48 @@
...
+  flex-basis: 0;
+}
+
+// places the connecting line with a pseudoclass
+.nonstop::before,
+auro-flight-segment::before,
+::slotted(*)::before {
+  content: "";
+  display: block;
+  background-color: #054687;
```
Comment (blackfalcon) [inline src/style-flightline.scss:43]:
```suggestion
  background-color: var(--auro-color-brand-atlas-500);
```

File: demo/demo.md [hunk 15]
```diff
@@ -1,56 +1,170 @@
...
 
-Nulla at augue facilisis `odio lobortis` molestie vitae a nulla.
+The flightline represents the layovers and stopovers the guest will take throughout their journey. For instance, flying from SEA (Seattle, WA) to AVP (Scranton, PA) will take a *layover* in ORD (Chicago, IL), while a guest flying from [ANC (Anchorage, AK) to ADK (Adak Island, AK) will have a *stopover* in CDB (Cold Bay, AK)](https://onemileatatime.com/alaska-airlines-adak-cold-bay/). They won't get off the aircraft in Cold Bay, kind of like a bus. We cannot predict stopover duration due to variants in passenger and cargo loads.
 
-## auro-flightline use cases
+Guests with physical disabilities appreciate indication of whether a flight will require them to deplane, and typically optimize for the most direct route. For this reason, [Change of Gauge flights](https://www.travelweekly.com/Mark-Pestronk/Agents-must-identify-change-of-gauge-services), or COG for short, require a *layover* indication even if they are filed as a direct flight. Please contact gus@alaskaair.com for clarification on this. 
 
-The `auro-flightline` element should be used in situations where users may:
+<h2>Nonstop (SEAEWR)</h2>
```
Comment (blackfalcon) [inline demo/demo.md:9]:
```suggestion
## Nonstop

The following illustrates a nonstop flight, for example, SEA to EWR.
```

File: demo/demo.md [hunk 16]
```diff
@@ -1,56 +1,170 @@
...
+1      8 F7 D7 P7 I7 U5*SEAEWR 066  730A  355P 73J S 0 /E
+         A2 Y7 Z7 S7 B7 M7 H7 Q7 L7 V7
+  ```
+</auro-accordion>
 
-|convallis|tristique|nisl dignissim|eleifend|
-|---|---|---|---|
-|√|√|||
-|||√|√|
+<h2>1 Connection (SEAAVP)</h2>
```
Comment (blackfalcon) [inline demo/demo.md:29]:
```suggestion
## Flight with a single connection

The following illustrates flight with one connection, for example, SEA to AVP.
```

File: demo/demo.md [hunk 17]
```diff
@@ -1,56 +1,170 @@
...
+ 10FEB  WED   SEA/PST     AVP/EST¥3
+1     34 F7 D7 P7 A0 Y7 Z7*SEAORD 080  810A  208P 73J S 0 /E
+         S7 B7 M7 H7 Q7 L7 V7 K7 G7
+2  *4121 Y7 S7 B7 M7 H7 Q7    AVP   N  455P  752P ERJ 0 XJS /E
+         L7 V7 K7 G7 T7 R7
+OPERATED BY /ENVOY AIR AS AMERICAN EAGLE
   ```
+</auro-accordion>
+
+<h2>1 Stopover (ANCADK)</h2>
```
Comment (blackfalcon) [inline demo/demo.md:55]:
```suggestion
## Flight with a single stopover

The following illustrates a flight with one stopover, for example, SEA to ADK.
```

File: demo/demo.md [hunk 18]
```diff
@@ -1,56 +1,170 @@
...
+ 14FEB  SUN   ANC/Z-9     ADK/HST-1
+1   184 F7 D7 P7 I7 U4 A2*ANCADK 000 -1 1000A 1255P 73G 1 WJ /E
+        Y7 Z7 S7 B7 M7 H7 Q7 L7 V7
+2   184 F7 D7 P7 I7 U6 A3*ANCADK 000  3¥1000A 1255P 73G 1 WJ /E
+        Y7 Z7 S7 B7 M7 H7 Q7 L7 V7
+  ```
 </auro-accordion>
 
-## Then there is more
+<h2>2 Stopovers (JNUANC)</h2>
```
Comment (blackfalcon) [inline demo/demo.md:80]: Delete this example

File: demo/demo.md [hunk 19]
```diff
@@ -1,56 +1,170 @@
...
+        Z4 S3 B3 M2 H2 Q2 L1 V0
+3    65 F7 D7 P7 I7 U4 A3 Y7*   PSG 033  705A 1051A 73G S 2 /E
+        Z7 S7 B7 M7 H7 Q7 L7 V7
+  ```
+</auro-accordion>
 
-toggleDialogClose = (elName) => {
-  let dialog = document.querySelector(elName);
-  const html = document.querySelector('html');
+<h2>Stopover + Layover (ADKSEA)</h2>
```
Comment (blackfalcon) [inline demo/demo.md:107]:
```suggestion
## Flight with a stopover and a layover

The following illustrates flight with a stopover and a layover, for example, ADK to SEA.
```

File: demo/demo.md [hunk 20]
```diff
@@ -1,56 +1,170 @@
...
+  ```
+   10FEB  WED   ADK/HST     SEA/PST¥2
+1   187 F7 D7 P7 I7 U4 A3*ADKANC 016    205P  725P 73H 1 WJ /E
+        Y7 Z7 S7 B7 M7 H7 Q7 L7 V7
+2   114 F1 D1 P0 I0 U0 A0*   SEA 100 1¥ 225A  650A 73G 0 /E
+        Y7 Z7 S7 B7 M7 H6 Q6 L5 V4
+  ```
+</auro-accordion>
+
+<h2>Stopover Layover International Flight (ADKBCN)</h2>
```
Comment (blackfalcon) [inline demo/demo.md:134]:
```suggestion
## Multi-leg flight with layovers and stopovers

The following illustrates an international flight with stopovers and layovers, for example, ADK to BCN.
```

**Review by blackfalcon — CHANGES_REQUESTED** [review]
Looks great. Please address the final content updates and we are home free.

File: demo/demo.md [hunk 21]
```diff
@@ -1,56 +1,170 @@
...
+<auro-accordion lowProfile justifyRight>
+  <span slot="trigger">See code</span>
 
-> Interdum et malesuada fames ac ante ipsum primis in faucibus. Nam fermentum libero ipsum, ac tempor sapien blandit in. Nam tincidunt non felis molestie varius.
+  ```html
+  <auro-flightline>
+  </auro-flightline>
+  ```
+  ```
+   10FEB  WED   SEA/PST     EWR/EST¥3
```
[the remark this answers is not in the document]
Comment (gusnaughton) [reply demo/demo.md:23]: This is Sabre output intended for use of TPOs validating their scenario in production against our intended designs. I think this is useful and each of these availability blocks are unique to the individual demo. I think they should remain coupled - thoughts?

File: demo/demo.md (hunk 13 above)
Comment (gusnaughton) [reply demo/demo.md:7]: This is a legal requirement that we need to comply to. Do we want to risk engineers not travelling to that supporting document and slipping out of compliance?

File: demo/demo.md (hunk 20 above)
Comment (gusnaughton) [reply demo/demo.md:134]: Multi-leg flights do not exist

File: docs/usecases.md [hunk 22]
```diff
@@ -0,0 +1,158 @@
+# Business Cases
+Below are example markets searchable in ITA and Sabre for availability.
+
+For more information about the difference between a layover and a stopover,
```
Comment (blackfalcon) [inline docs/usecases.md:4]:
```suggestion
For more information, [please see this article](https://www.turkishairlines.com/en-int/any-questions/what-is-the-difference-between-a-layover-and-a-stopover/) about the differences between a layover and a stopover.
```

File: docs/usecases.md [hunk 23]
```diff
@@ -0,0 +1,158 @@
...
+[please see this article.](https://www.turkishairlines.com/en-int/any-questions/what-is-the-difference-between-a-layover-and-a-stopover/)
```
Comment (blackfalcon) [inline docs/usecases.md:5]: Delete line

File: docs/usecases.md [hunk 24]
```diff
@@ -0,0 +1,158 @@
...
+  <auro-flightline></auro-flightline>
+</div>
+
+<auro-accordion lowProfile justifyRight>
+  <span slot="trigger">See code</span>
+
+  ```html
+  <auro-flightline></auro-flightline>
+  ```
+  ```
```
Comment (blackfalcon) [inline docs/usecases.md:20]: All these blobs of data should have a preceding line of text that describes what it is.

File: docs/usecases.md [hunk 25]
```diff
@@ -0,0 +1,158 @@
...
+</div>
+
+<auro-accordion lowProfile justifyRight>
+  <span slot="trigger">See code</span>
+
+  ```html
+  <auro-flightline>
+    <auro-flight-segment iata="LAX"></auro-flight-segment>
+  </auro-flightline>
+  ```dion>
```
Comment (blackfalcon) [inline docs/usecases.md:42]: This is broken

File: docs/usecases.md [hunk 26]
```diff
@@ -0,0 +1,158 @@
...
+ 10FEB  WED   SEA/PST     AVP/EST¥3
+1     34 F7 D7 P7 A0 Y7 Z7*SEAORD 080  810A  208P 73J S 0 /E
+         S7 B7 M7 H7 Q7 L7 V7 K7 G7
+2  *4121 Y7 S7 B7 M7 H7 Q7    AVP   N  455P  752P ERJ 0 XJS /E
+         L7 V7 K7 G7 T7 R7
+OPERATED BY /ENVOY AIR AS AMERICAN EAGLE
+  ```
+</auro-accordion>
+
+## Flight with a single stopover (ANCADK)
```
Comment (blackfalcon) [inline docs/usecases.md:70]: Is this common to be written like this?  `(ANCADK)`?

File: demo/demo.md [hunk 27]
```diff
@@ -1,56 +1,105 @@
...
+# auro-Flightline
 
-Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nullam convallis in tellus nec pellentesque. Integer bibendum ligula sit amet vehicula gravida. Maecenas accumsan, ligula vitae molestie iaculis, tellus mi laoreet ex [install instructions](https://auro.alaskaair.com/components/auro/button/install), ac malesuada velit dolor vel mi. Cras et rutrum urna. Sed mattis mi eu tortor ullamcorper, egestas bibendum mauris cursus. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Phasellus viverra eros eget neque commodo vulputate. In tempus eu velit at dictum.
+auro-flightline represents any layovers and/or stopovers a guest may encounter throughout their journey. 
 
-Nulla at augue facilisis `odio lobortis` molestie vitae a nulla.
+For mobile, the number of stops is automatically calculated by the number of layovers and/or stopovers added in the DOM via the `auro-flight-segment` element. 
 
-## auro-flightline use cases
+## Component use cases
```
Comment (blackfalcon) [inline demo/demo.md:7]: Insert line break between this header and the next content line. It's a markdown thing.

File: demo/demo.md [hunk 28]
```diff
@@ -1,56 +1,105 @@
...
+Use the auro-flightline element to:
+* Flying from SEA (Seattle, WA) to AVP (Scranton, PA) will require a *layover* in ORD (Chicago, IL)
+* Flying from [ANC (Anchorage, AK) to ADK (Adak Island, AK) will have a *stopover* in CDB (Cold Bay, AK)](https://onemileatatime.com/alaska-airlines-adak-cold-bay/). These passengers will not deplane in Cold Bay.
+Stopover duration cannot be predicted due to variants in passenger and cargo loads.
 
-The `auro-flightline` element should be used in situations where users may:
+## Nonstop
```
Comment (blackfalcon) [inline demo/demo.md:13]: Please add suggested descriptive text here.

File: demo/demo.md [hunk 29]
```diff
@@ -1,56 +1,105 @@
...
+  ```html
+  <auro-flightline></auro-flightline>
+  ```
+</auro-accordion>
 
-|convallis|tristique|nisl dignissim|eleifend|
-|---|---|---|---|
-|√|√|||
-|||√|√|
+## Flight with a single layover
```
Comment (blackfalcon) [inline demo/demo.md:26]: Please add suggested descriptive text here.

File: demo/demo.md [hunk 30]
```diff
@@ -1,56 +1,105 @@
...
 
   ```html
-  <auro-flightline cssClass="testClass">Hello World!</auro-flightline>
+  <auro-flightline>
+    <auro-flight-segment iata="ORD" duration="3h 40m"></auro-flight-segment>
+  </auro-flightline>
   ```
+</auro-accordion>
+
+## Flight with a single stopover
```
Comment (blackfalcon) [inline demo/demo.md:44]: Please add suggested descriptive text here.

File: demo/demo.md (hunk 13 above)
Comment (blackfalcon) [reply demo/demo.md]: What is the legal requirement? That is not specified here. I see `Guests with physical disabilities appreciate indication...` and then descriptive text. 

If there is a legal requirement, say that. Be specific and direct. 

Also, never use your personal email address as a way for people to ask questions. That is why we have issues and other channels of communication. While you are the dev building this element, you are not the single point of contact for any and all issues that may arise from this tool.

File: docs/usecases.md [hunk 31]
```diff
@@ -0,0 +1,158 @@
+# Business Cases
```
Comment (blackfalcon) [inline docs/usecases.md:1]:
```suggestion
# Airline use specifications
```

I feel that this should be more direct. Emphasize that there are guidelines to be followed and that there are real repercussions for not adhering to them.

File: docs/api.md (hunk 11 above)
[the remark this answers is not in the document]
Comment (gusnaughton) [reply docs/api.md]: Yes - we currently use this when we collapse down to responsive mode '2 stops'.

File: docs/usecases.md (hunk 26 above)
Comment (gusnaughton) [reply docs/usecases.md:70]: Typically the parenthesis are left out, but yes - this is Sabre and ITA terminology for describing a citypair.

**Review by blackfalcon — APPROVED** [review]

**Pull request merged**